pkg/scheduler/internal/queue/scheduling_queue.go

```diff
@@ -908,12 +925,11 @@ func (p *PriorityQueue) movePodsToActiveOrBackoffQueue(podInfoList []*framework.
 	// AddUnschedulableIfNotPresent we need to know whether events where
 	// observed while processing them.
 	for uid, inFlight := range p.inFlightPods {
-		if !p.runRequeuePlugins(ctx, inFlight.pod, &event, oldObj, newObj) {
-			// This Pod isn't worth putting to activeQ or backoffQ.
-			continue
-		}
-		klog.V(6).InfoS("In-flight pod with concurrent event", "pod", klog.KObj(inFlight.pod.Pod), "event", event.Label, "schedulingCycle", p.schedulingCycle)
+		successPlugins := p.runRequeuePlugins(ctx, inFlight.pod, &event, oldObj, newObj)
+
+		klog.V(6).InfoS("In-flight pod with concurrent event", "pod", klog.KObj(inFlight.pod), "event", event.Label, "schedulingCycle", p.schedulingCycle)
 		inFlight.moveRequestCycle = p.schedulingCycle
+		inFlight.successRequeuePlugins = inFlight.successRequeuePlugins.Union(successPlugins)
```

Comment:
Danger!

Multiple different events may get handled at the same time, therefore access to this field must be protected.

Reply:
Oh, OK. Until now, I thought the event handler processes each event one by one.

Reply:
But, `p.lock.Lock()` is already done in the caller fn. Do we need to do something else to protect this field?

Reply:
You are right, this should be enough because reading it holds the same lock.